app/controllers/songs_controller.rb

```diff
@@ -1,10 +1,25 @@
 class SongsController < ApplicationController
   def index
+    @songs = Song.all
   end
 
   def new
   end
 
   def edit
   end
+  
+  def search
+    @results = true
+    if params[:search].nil?
+      @songs = Song.all
+    else
+      @songs = Song.search params[:search]
```

Comment:
Query logic shouldn't be contained in the model generally. After all, it makes sense models wouldn't know how to retrieve themselves.